### PR TITLE
Added the title in the submit action

### PR DESCRIPTION
**Pull request by IbrahimSulai**
Added the title as a parameter in onExecuteAction for the submit action. 

Cross verified and confirmed the same action in native platforms.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5217)

### CODE REVIEW

**Review by vivekvijayakrishnan — APPROVED** [review]
LGTM

File: source/community/reactnative/src/components/actions/action-button.js [hunk 1]
```diff
@@ -114,7 +114,8 @@ export class ActionButton extends React.Component {
 			else
 				mergedObject["actionData"] = this.data;
 		}
-		let actionObject = { "type": this.payload.type, "data": mergedObject };
+		const title = this.payload.title
+		let actionObject = { "type": this.payload.type,"title":Utils.isNullOrEmpty(title) ? "" : title, "data": mergedObject };
 		this.onExecuteAction(actionObject, this.payload.ignoreInputValidation === true);
```
Comment (BalajiR) [inline source/community/reactnative/src/components/actions/action-button.js:119]: If `ignoreInputValidation` is Boolean, do we need this explicit check here ?

File: source/community/reactnative/src/components/actions/action-button.js [hunk 2]
```diff
@@ -114,7 +114,8 @@ export class ActionButton extends React.Component {
 			else
 				mergedObject["actionData"] = this.data;
 		}
-		let actionObject = { "type": this.payload.type, "data": mergedObject };
+		const title = this.payload.title
```
Comment (BalajiR) [inline source/community/reactnative/src/components/actions/action-button.js:117]: Shall we have :
`const { title = "",  ignoreInputValidation } = this.payload;`
This also avoid the null check function call on `title`

File: source/community/reactnative/src/components/actions/select-Action.js [hunk 3]
```diff
@@ -32,7 +32,8 @@ export class SelectAction extends React.Component {
 	onClickHandle() {
 		switch (this.payload.type) {
 			case Constants.ActionSubmit:
-				let actionObject = { "type": Constants.ActionSubmit, "data": this.props.selectActionData.data };
+				const title = this.props.selectActionData.title
```
Comment (BalajiR) [inline source/community/reactnative/src/components/actions/select-Action.js:35]: Shall we read the properties by destructuring the payload for better readability.

**FROM**
```
const title = this.props.selectActionData.title
let actionObject = { "type": Constants.ActionSubmit, "title":Utils.isNullOrEmpty(title) ? "" : title, "data": this.props.selectActionData.data };
this.onExecuteAction(actionObject);
```
**TO**
```
const { type, title = "", data } = selectActionData;
this.onExecuteAction({type, title, data})
```

**Review by BalajiR — CHANGES_REQUESTED** [review]

File: source/community/reactnative/src/components/actions/action-button.js (hunk 1 above)
Comment (IbrahimSulai) [reply source/community/reactnative/src/components/actions/action-button.js]: Resolved

File: source/community/reactnative/src/components/actions/action-button.js (hunk 2 above)
Comment (IbrahimSulai) [reply source/community/reactnative/src/components/actions/action-button.js]: Updated

File: source/community/reactnative/src/components/actions/select-Action.js (hunk 3 above)
Comment (IbrahimSulai) [reply source/community/reactnative/src/components/actions/select-Action.js]: Resolved

**Review by BalajiR — APPROVED** [review]
LGTM

**Pull request merged**